### DISCUSSION TITLE
Order of tasks in GroupAggregation

### DISCUSSION THREAD

**Discussion by PolamarasettyRohitha** [hibiken/asynqmon#279] [Q&A]
Group Aggregation is **not preserving** the order of tasks.

I enqueued few tasks under same group name, when they are executed, order of execution of those tasks was different from the order that were enqueued.

example:  
Order of enqueue of tasks in a group:
      {"UserID":0,"TemplateID":"some:template:id"}
      {"UserID":1,"TemplateID":"some:template:id"}
      {"UserID":2,"TemplateID":"some:template:id"}
      {"UserID":3,"TemplateID":"some:template:id"}
      {"UserID":4,"TemplateID":"some:template:id"}
      {"UserID":5,"TemplateID":"some:template:id"}

when I process those tasks, the order of their execution is :
    {"UserID":1,"TemplateID":"some:template:id"}
    {"UserID":5,"TemplateID":"some:template:id"}
    {"UserID":0,"TemplateID":"some:template:id"}
    {"UserID":3,"TemplateID":"some:template:id"}
    {"UserID":2,"TemplateID":"some:template:id"}
    {"UserID":4,"TemplateID":"some:template:id"}

Is there a way to preserve order of tasks those are grouped.

Thank you.

**Comment by IsaacDSC** [comment]
Would it be good task group use concurrent process ?
Do you suggest creating a feature to ensure the order in which this message request is publisher ?

cc: @PolamarasettyRohitha
